packages/common/infra/src/sync/job/runner.ts

```diff
@@ -0,0 +1,60 @@
+import { DebugLogger } from '@affine/debug';
+
+import { MANUALLY_STOP, throwIfAborted } from '../../utils';
+import type { Job, JobQueue } from './queue';
+
+const logger = new DebugLogger('job-runner');
+
+export class JobRunner<J> {
+  abort: AbortController | null = null;
+
+  constructor(
+    private readonly queue: JobQueue<J>,
+    private readonly worker: (
+      jobs: Job<J>[],
+      signal: AbortSignal
+    ) => Promise<void>,
+    private readonly interval: () => Promise<void> = async () => {}
+  ) {}
+
+  start() {
+    this.stop();
```

Comment:
puzzling name, better to be something like `listenOnStopSignal`

Reply:
which name?